zhaquirks/tuya/ts011f_plug.py

```diff
@@ -1022,3 +1032,51 @@ class Plug_4AC_2USB_Metering(EnchantedDevice):
             },
         },
     }
+
+
+class Plug_v2(CustomDevice):
+    """Another TS011F Tuya plug."""
+
+    signature = {
+        MODEL: "TS011F",
+        ENDPOINTS: {
+            # "profile_id": 260,
+            # "device_type": "0x0051",
+            # "in_clusters": ["0x0000", "0x0003", "0x0004", "0x0005", "0x0006", "0x0702", "0x0b04", "0xe001"],
+            # "out_clusters": []
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Identify.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    OnOff.cluster_id,
+                    Metering.cluster_id,
+                    ElectricalMeasurement.cluster_id,
+                    TuyaZBExternalSwitchTypeCluster.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [],
+            },
+        },
+    }
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.ON_OFF_PLUG_IN_UNIT,
```

Comment:
Looks nice and the unit of measurement is being set but way is you changing the device type ?
For my smart plug is more advanced then one simple on-off plug unit and some devices dont like changing the device type and getting problems.

Reply:
This approach was recommended by @javicalle here https://github.com/zigpy/zha-device-handlers/issues/1764#issuecomment-1272390644. I've just used this quirk and added the unit_of_measurement. I don't know the background to why this was recommended, but happy to change it and retest, I have no idea what to look for to see which is the right approach however.

Reply:
Can you doing one test with `zha.DeviceType.SMART_PLUG` as in the signature and see if its working OK ?

The `Fixes:` looks only linking the first issue linked.
Its must being "`Fixes:` #XXXX,  `Fixes:` #XXXX " so i was having half wrong :-((

Reply:
Yeah, I believe just ``zha.DeviceType.SMART_PLUG`` should work. Would be nice if you could test it with that profile type.

(Also linked both issues correctly to this PR now)

Reply:
This does work fine!